### PR TITLE
LP-2025 Create smart referral model for invitation

### PR DESCRIPTION
**Pull request by harris-irfan**
Register smart referral app
Register model with admin
Add unit tests

### Description

[LP-2025](https://philanthropyu.atlassian.net/browse/LP-2025)

### CODE REVIEW

File: openedx/features/smart_referral/models.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.contrib.auth.models import User
+from django.db import models
+from model_utils.models import TimeStampedModel
+
+
+class SmartReferral(TimeStampedModel):
```
Comment (zahid-arbisoft) [inline openedx/features/smart_referral/models.py:9]: Lets add `__unicode__()` and return something like `{username} reffered {contact_email}`

File: openedx/features/smart_referral/apps.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.apps import AppConfig
+
+
+class MarketplaceConfig(AppConfig):
```
Comment (muak50) [inline openedx/features/smart_referral/apps.py:7]: Won't that be `SmartReferralConfig` as the app directory and below name field says

File: openedx/features/smart_referral/tests/test_models.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    def test_save_smart_referral_model_with_mandatory_fields_successfully(self):
+        """
+        Create a referral, and assert that it is saved successfully with all mandatory data
+        """
+        referral = SmartReferralFactory()
+
+        referral_from_database = SmartReferral.objects.get(pk=referral.id)
+
+        self.assertIsNotNone(referral_from_database)
+        self.assertEqual(referral.contact_email, referral.contact_email)
```
Comment (mabdullahz) [inline openedx/features/smart_referral/tests/test_models.py:25]: `self.assertEqual(referral_from_database.contact_email, referral.contact_email)`

File: openedx/features/smart_referral/apps.py (hunk 2 above)
Comment (zahid-arbisoft) [reply openedx/features/smart_referral/apps.py]: Good catch @UsmanArbisoft

**Pull request merged**